tests/memory-report.cpp

```
@@ -0,0 +1,349 @@
+        "  %-29s resident=%s, private/commit=%s\n",
+        "os observed peak",
+        formatBytes(observedPeak.residentBytes).c_str(),
+        formatBytes(observedPeak.commitBytes).c_str()
+    );
+    if (sampledPeakSnapshot && !sampledPeakSnapshot->label.empty())
+    {
+        std::printf("  highest sampled point: %s\n", sampledPeakSnapshot->label.c_str());
+    }
+
+    for (const MemoryReportSnapshot& snapshot : gMemoryReportSnapshots)
+    {
```

Cppcheck (2.21.0)

[style] The function 'attach' is never used.

(unusedFunction)